oxlate/headers.py

```diff
@@ -0,0 +1,38 @@
+from .utils import string_to_case_number, case_number_to_string
+
+class Headers:
+    def __init__(self):
+        self._data = {}
+
+    def add(self, name, value, adjust_case_to_allow_duplicates=False):
+        # import ipdb; ipdb.set_trace()
+        if adjust_case_to_allow_duplicates:
+            if self._data.get(name):
+                # Collision, need to change case
+                same_names_case_insensitive = list(filter(lambda key: key.lower() == name.lower(), self._data.keys()))
+                existing_names_case_binary_representations = set(map(string_to_case_number, same_names_case_insensitive))
+                available_case_binary_representations = sorted(set(range(pow(2,len(name)))) - existing_names_case_binary_representations)
```

Comment:
I think this formula might be incorrect.  2^N works if every character has upper- and lower-case options, but `-` (as in `set-cookie`) does not.  So really it should be 2^A where A is the number of alpha (a-z) characters.

Reply:
This code is also exponentially slow in `len(name)` - that could get ugly fast.

Reply:
fixed